.github/workflows/ci.yaml

```diff
@@ -142,3 +142,39 @@ jobs:
         with:
           name: docs
           path: docs/_build/html
+
+
+  api-backward-compatibilty:
+    name: Run Backward Compatibility Checks
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    steps:
+      - id: base_branch
+        run: |
+            echo "::set-output name=branch::$(echo ${GITHUB_BASE_REF#refs/heads/})"
+      - name: Checkout Base Branch
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ steps.base_branch.outputs.branch }}
+          path: base/mlrun
+      - name: Checkout Head Branch and Stable with the target Branch
+        run : |
+          mkdir head && cd head
+          git init
+          git config --global user.name github-actions
+          git config --global user.email github-actions@mlrun
+          git config --global github.token ${{ secrets.GITHUB_TOKEN }}
+          git config --global pull.rebase false
+          git clone $GITHUB_SERVER_URL/$GITHUB_ACTOR/mlrun.git
+          cd mlrun
+          git remote add upstream $GITHUB_SERVER_URL/$GITHUB_REPOSITORY
+          git checkout $GITHUB_HEAD_REF
+          git pull upstream $GITHUB_BASE_REF --commit
```

Comment:
Why not using the `actions/checkout` to get the code, and then just do the `git pull ...` in another step
Also, are you sure the config and such are needed ? it's an open source repository, why would a token need to be defined in order to just pull from it ?